foursight_core/app_utils.py

```diff
@@ -467,7 +863,199 @@ def view_foursight(self, environ, is_admin=False, domain="", context="/"):
+            bucket_names=bucket_names,
+            environment_and_bucket_info=environment_and_bucket_info,
+            declared_data=declared_data,
+            identity_name=gac_name,
+            identity_secrets=gac_values,
+            resources=resources,
+            ping_portal=self.ping_portal(environ, stage_name),
+            ping_elasticsearch=self.ping_elasticsearch(environ),
+            ping_sqs=self.ping_sqs(),
+            versions=versions,
+            os_environ=os_environ
+        )
+        html_resp.status_code = 200
+        return self.process_response(html_resp)
+
+    def view_user(self, request, environ, is_admin=False, domain="", context="/", email=None):
+        html_resp = Response('Foursight viewing suite')
+        html_resp.headers = {'Content-Type': 'text/html'}
+        request_dict = request.to_dict()
+        stage_name = self.stage.get_stage()
+        users = []
+        for this_email in email.split(","):
+            try:
+                this_user = ff_utils.get_metadata('users/' + this_email.lower(),
+                                                  ff_env=full_env_name(environ), add_on='frame=object')
+                users.append({"email": this_email, "record": this_user})
+            except Exception as e:
+                users.append({"email": this_email, "record": {"error": str(e)}})
+        template = self.jin_env.get_template('user.html')
+        html_resp.body = template.render(
+            request=request,
+            version=self.get_app_version(),
+            package=self.APP_PACKAGE_NAME,
+            env=environ,
+            domain=domain,
+            context=context,
+            environments=self.get_unique_full_environment_names(),
+            stage=stage_name,
+            is_admin=is_admin,
+            is_running_locally=self.is_running_locally(request_dict),
+            logged_in_as=self.get_logged_in_user_info_from_jwt_token_cookie(request_dict),
+            users=users,
+            auth0_client_id=self.get_auth0_client_id(environ, stage_name),
+            aws_account_number=self.get_aws_account_number(),
+            main_title=self.html_main_title,
+            favicon=self.get_favicon(),
+            load_time=self.get_load_time(),
+            init_load_time=self.init_load_time,
+            lambda_deployed_time=self.get_lambda_last_modified(),
+            running_checks='0',
+            queued_checks='0'
+        )
+        html_resp.status_code = 200
+        return self.process_response(html_resp)
+
+    def view_users(self, request, environ, is_admin=False, domain="", context="/"):
+        html_resp = Response('Foursight viewing suite')
+        html_resp.headers = {'Content-Type': 'text/html'}
+        request_dict = request.to_dict()
+        stage_name = self.stage.get_stage()
+        users = []
+        user_records = ff_utils.get_metadata('users/', ff_env=full_env_name(environ), add_on='frame=object&limit=10000')
```

Comment:
This should do paging; will get to it at some point.

Reply:
Add a TODO comment?